packages/app/src/components/404/Custom404Page.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import { Box, Button, Divider, Grid, Typography, makeStyles } from '@material-ui/core';
+import { Content, Page } from '@backstage/core-components';
+import { SearchBar, SearchContextProvider, SearchPagination } from '@backstage/plugin-search-react';
+import { searchPlugin } from '@backstage/plugin-search';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+import { searchResultCustomList } from '../search/SearchResultCustomList';
+import { useRouteRef } from '@backstage/core-plugin-api';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+
+const useStyles = makeStyles(theme => ({
+  title: {
+    gap: theme.spacing(1),
+    display: 'grid',
+    alignItems: 'center',
+    gridTemplateColumns: '1fr auto',
+    '&> button': {
+      marginTop: theme.spacing(1),
+    },
+  },
+  input: {
+    flex: 1,
+  },
+  link: {
+    color: theme.palette.primary.main,
+  },
+  button: {
+    '&:hover': {
+      background: 'none',
+    },
+  },
+}));
+
+const rootRouteRef = searchPlugin.routes.root;
+
+export function Custom404Page() {
+  const classes = useStyles();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const searchRootRoute = useRouteRef(rootRouteRef)();
+  const searchBarRef = useRef<HTMLInputElement | null>(null);
+
+  // grab the url slug
+  const pagePath = location.pathname.split('/').pop();
+
+  const preFiltered = {
+    term: pagePath?.replaceAll('-', ' ')?.replaceAll('_', ' ') ?? '',
+    types: [],
+    filters: {},
+  };
+
+  useEffect(() => {
+    searchBarRef?.current?.focus();
+  });
+
+  // This handler is called when "enter" is pressed
+  const handleSearchBarSubmit = useCallback(() => {
+    // Using ref to get the current field value without waiting for a query debounce
+    const query = searchBarRef.current?.value ?? '';
+    navigate(`${searchRootRoute}?query=${query}`);
+  }, [navigate, searchBarRef]);
+
+  return (
+    <Page themeId="home">
+      <Content>
+        <SearchContextProvider initialState={preFiltered}>
+          <Box padding="calc(2.1rem - 24px) 9%">
+            <Typography variant="h1">Page not found</Typography>
+            <Divider orientation="horizontal" variant="middle" />
+            <Box pt={3}>
+              <Typography variant="h6">
+                <Link className={classes.link} to="#" onClick={() => navigate(-1)}>Go back</Link> or&nbsp;
+                <Link className={classes.link} to={`https://classic.developer.gov.bc.ca/?q=${preFiltered.term}`}>Search Classic DevHub</Link> - please&nbsp;
```

Comment:
are we doing the tracking we needed on this link so we can see the offsite clicks from here? I remember this wasn't easy...

Reply:
I've just double checked this and click tracking is working on this page. It doesn't seem to work in the PR preview deployment. But I can see the events firing via the Chrome extension in my local dev environment.